onnxruntime/core/session/provider_bridge_ort.cc

```diff
@@ -1195,43 +1170,28 @@ OrtTensorRTProviderOptionsV2 OrtTensorRTProviderOptionsToOrtTensorRTProviderOpti
 
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Tensorrt(const OrtTensorRTProviderOptions* provider_options) {
   OrtTensorRTProviderOptionsV2 trt_options_converted = onnxruntime::OrtTensorRTProviderOptionsToOrtTensorRTProviderOptionsV2(provider_options);
-  if (auto* provider = s_library_tensorrt.Get())
-    return provider->CreateExecutionProviderFactory(&trt_options_converted);
-
-  return nullptr;
+  return s_library_tensorrt.Get().CreateExecutionProviderFactory(&trt_options_converted);
 }
 
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Tensorrt(const OrtTensorRTProviderOptionsV2* provider_options) {
-  if (auto* provider = s_library_tensorrt.Get())
-    return provider->CreateExecutionProviderFactory(provider_options);
-
-  return nullptr;
+  return s_library_tensorrt.Get().CreateExecutionProviderFactory(provider_options);
 }
 
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_MIGraphX(const OrtMIGraphXProviderOptions* provider_options) {
-  if (auto* provider = s_library_migraphx.Get())
-    return provider->CreateExecutionProviderFactory(provider_options);
-
-  return nullptr;
+  return s_library_migraphx.Get().CreateExecutionProviderFactory(provider_options);
 }
 
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_OpenVINO(const OrtOpenVINOProviderOptions* provider_options) {
-  if (auto* provider = s_library_openvino.Get())
-    return provider->CreateExecutionProviderFactory(provider_options);
-
-  return nullptr;
+  return s_library_openvino.Get().CreateExecutionProviderFactory(provider_options);
 }
 
 ProviderInfo_OpenVINO* GetProviderInfo_OpenVINO() {
-  if (auto* provider = s_library_openvino.Get())
-    return reinterpret_cast<ProviderInfo_OpenVINO*>(provider->GetInfo());
-  return nullptr;
+  return reinterpret_cast<ProviderInfo_OpenVINO*>(s_library_openvino.Get().GetInfo());
 }
 
-ProviderInfo_CUDA* TryGetProviderInfo_CUDA() {
-  if (auto* provider = s_library_cuda.Get())
-    return reinterpret_cast<ProviderInfo_CUDA*>(provider->GetInfo());
-
+ProviderInfo_CUDA* TryGetProviderInfo_CUDA() try {
+  return reinterpret_cast<ProviderInfo_CUDA*>(s_library_cuda.Get().GetInfo());
+} catch (...) {
```

Comment:
>catch (...) [](http://example.com/codeflow?start=2&length=11)

do we want to swallow all exceptions here? if so, would be good to have a comment explaining why

Reply:
Yeah, it's not ideal but the function should return 'nullptr' if it can't load and a valid pointer if it can. Throwing an exception would require rethinking all of the calling code. We do write a log when throwing an exception at least.

Reply:
maybe log it here too? I'm not aware of exceptions generally being logged when thrown

Reply:
Done